Gems/Terrain/Assets/Shaders/Terrain/TerrainSrg.azsli

```diff
@@ -13,6 +13,8 @@
 static const uint MacroClipmapStackSizeMax = 16u;
 static const uint DetailClipmapStackSizeMax = 16u;
 static const uint StackSizeMax = 16u; // = max(MacroClipmapStackSizeMax, DetailClipmapStackSizeMax)
+static const uint MacroClipmapStackSizeMax_x6 = 96u;
+static const uint DetailClipmapStackSizeMax_x6 = 96u;
```

Comment:
Could you do `= MacroClipmapStackSizeMax * 6u;`? Also, this would again benefit from something explaining that this is related to having 6 regions (I'm guessing), and why there are 6 regions.

Might even consider a name based on the regions, not the number of regions, like 'MacroClipmapRegionStackSizeMax'.

Reply:
I think this is a symptom of trying to make a single compute shader that does all the updating for every clipmap in every possible region all at once. This could be simplified so each invocation of the compute shader only cares about a single contiguous region on the clipmap.

Reply:
Unfortunately, the shader compiler is not smart enough to figure out those constant sizes. Both `static const uint size = MacroClipmapStackSizeMax * 6u; array[size];` or `array[MacroClipmapStackSizeMax * 6u];` will fail at unresolved array dimensions.